doc/source/whatsnew/v1.2.0.rst

```diff
@@ -54,6 +54,7 @@ For example:
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
 - :class:`Index` with object dtype supports division and multiplication (:issue:`34160`)
+- :meth:`factorize` now supports ``na_sentinel=None`` to include NaN in the uniques of the values and remove ``dropna`` keyword from :meth:`factorize` (:issue:`35667`)
```

Comment:
maybe could clarify that the dropna was unintentionally added to the public facing api in 1.1 and backport this PR. wdyt?

Reply:
yep, agree!

Reply:
should this still be in 1.2 whatsnew btw?

Reply:
move to 1.1.1 whatsnew. we have a bugfix and regression section. I don't think this fits neatly in either, so could put it above both and maybe a bit more narrative than a regular change note.

Reply:
you could also frame it as an enhancement to pd.factorize that wasn't documented.

Reply:
thanks @simonjayhawkins i added an `other` section like we have in other whatsnew, how does it sound?

Reply:
looks good